datafusion/sqllogictest/test_files/order.slt

```diff
@@ -578,3 +578,192 @@ SortPreservingMergeExec: [log_c12_base_c11@0 DESC]
 
 statement ok
 drop table aggregate_test_100;
+
+
+# Sort with lots of repetition values
+# Test sorting a parquet file with 2 million records that has lots of values that are repeated
+statement ok
+CREATE EXTERNAL TABLE repeat_much STORED AS PARQUET LOCATION '../../parquet-testing/data/repeat_much.snappy.parquet';
+
+query I
+SELECT a FROM repeat_much ORDER BY a LIMIT 20;
+----
+2450962
+2450962
+2450962
+2450962
+2450962
+2450962
+2450962
+2450962
+2450962
+2450962
+2450962
+2450962
+2450962
+2450962
+2450962
+2450962
+2450962
+2450962
+2450962
+2450962
+
+
+# Create external table with optional pre-known sort order
+# DataFusion may take advantage of this ordering to omit sorts or use more efficient algorithms.
+statement ok
+set datafusion.catalog.information_schema = true;
+
+statement ok
+CREATE EXTERNAL TABLE dt (a_id integer, a_str string, a_bool boolean) STORED AS CSV WITH ORDER (a_id ASC) LOCATION 'file://path/to/table';
+
+#TODO: How to check for order in sqllogictest?
```

Comment:
Here I have a question: How can I check for pre-sorting with pure SQL? 

This is what the Rust unit test would call but I couldn't find anything in the information schema that would return the `file_sort_order` setting.
https://github.com/apache/arrow-datafusion/blob/be361fdd8079a2f44da70f6af6e9d8eb3f7d0020/datafusion/core/tests/sql/order.rs#L60-L62

Reply:
One way to test this is to run an explain plan that requires the order. The test should that the data is not re-sorted

For example:

```
# ORDER by query should be able to take advantage of the pre-existing order
# expect to not see SortExec
query EXPLAIN SELECT * FROM dt ORDER BY a_id ASC
```

Reply:
Thanks a lot for the fast review and the tip! 

I had to switch the table to an existing query since the old unit test with the inexistent `LOCATION 'file://path/to/table'` would only generate a logical plan, not the physical plan.

So I changed to a TPCH table, without a `WITH ORDER (o_orderkey ASC)` clause the physical plan shows a SortExec:
```
query TT
EXPLAIN SELECT o_orderkey, o_orderstatus FROM orders ORDER BY o_orderkey ASC
----
logical_plan
Sort: orders.o_orderkey ASC NULLS LAST
--TableScan: orders projection=[o_orderkey, o_orderstatus]
physical_plan
SortExec: expr=[o_orderkey@0 ASC NULLS LAST]
--CsvExec: file_groups={1 group: [[WORKSPACE_ROOT/datafusion/core/tests/tpch-csv/orders.csv]]}, projection=[o_orderkey, o_orderstatus], has_header=true
```

Adding it drops it from the physical plan and pushes it into output_ordering in the CsvExec:
```
query TT
EXPLAIN SELECT o_orderkey, o_orderstatus FROM orders ORDER BY o_orderkey ASC
----
logical_plan
Sort: orders.o_orderkey ASC NULLS LAST
--TableScan: orders projection=[o_orderkey, o_orderstatus]
physical_plan CsvExec: file_groups={1 group: [[WORKSPACE_ROOT/datafusion/core/tests/tpch-csv/orders.csv]]}, projection=[o_orderkey, o_orderstatus], output_ordering=[o_orderkey@0 ASC NULLS LAST], has_header=true
```